apps/src/code-studio/initApp/project.js

```diff
@@ -170,6 +177,32 @@ var projects = module.exports = {
     return url + fragment + queryString;
   },
 
+  /**
+   * Returns a share URL for the current project.
+   *
+   * Share URLs can vary by application environment and project type.  For most
+   * project types the share URL is the same as the project edit and view URLs,
+   * but has no action appended to the project's channel ID. Weblab is a special
+   * case right now, because it shares projects to codeprojects.org.
+   *
+   * This function depends on the document location to determine the current
+   * application environment.
+   *
+   * @returns {string} Fully-qualified share URL for the current project.
+   */
+  getShareUrl() {
+    const location = this.getLocation();
+    if (this.isWebLab()) {
+      const re = /([-.]?studio)?\.?code.org/i;
+      const environmentKey = location.hostname.replace(re, '');
+      const subdomain = environmentKey.length > 0 ? `${environmentKey}.` : '';
+      const port = 'localhost' === environmentKey ? `:${location.port}` : '';
```

Comment:
Gut says this kind of parsing could be more generically useful across our code.

Reply:
Think this should be extracted or reused somewhere right away?

Reply:
I think okay for now; but it would be good to keep an eye out for changes to use shared code going forward.  The Ruby side of the system has been somewhat good at using common code to deal with our various environments.